Clarify note for React Native users (#4512)
reactjs/reactjs.org: content/blog/2022-03-08-react-18-upgrade-guide.md

Please [report any issues](https://github.com/facebook/react/issues/new/choose) you encounter while upgrading to React 18.

> *Note for React Native users: React 18 will ship in a future version of React Native. Moreover, React 18 will rely on the New React Native Architecture to benefit from the new capabilities presented in this blogpost. For more information, see the [React Conf keynote here](https://www.youtube.com/watch?v=FZ0cG47msEk&t=1530s).*

Reviewer: ```suggestion
*Note for React Native users: React 18 will ship in a future version of React Native. This is because React 18 relies on the New React Native Architecture to benefit from the new capabilities presented in this blogpost. For more information, see the [React Conf keynote here](https://www.youtube.com/watch?v=FZ0cG47msEk&t=1530s).*
```
